### PR TITLE
BAVL-123 changes to send email notifications after successful court booking creation.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt [hunk 1]
```diff
@@ -32,4 +32,6 @@ class PrisonerValidator(val prisonerSearchClient: PrisonerSearchClient) {
 data class Prisoner(
   val prisonerNumber: String,
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt:33]: Isn't there a model for prisoner? OffenderDetail - something like that?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  val prisonId: Long = 0,
+
+  val code: String,
+
+  val description: String,
+
+  val enabled: Boolean,
+
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt:23]: Worth a comment on the enabled attrib?  Enabled == Courts/probation cannot self-serve, but we will accept bookings from the prison (via DPS and A&A). Maybe enabled is the wrong name?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt [hunk 3]
```diff
@@ -19,13 +20,28 @@ class BookingFacade(
...
 
-  private fun sendNewCourtBookingEmail(newBooking: VideoBooking) {
+    return booking.videoBookingId
+  }
+
+  private fun sendNewCourtBookingEmail(booking: VideoBooking, prisoner: Prisoner) {
     log.info("TODO - send new court booking email.")
+
+    // Agreed with Tim will be using a ContactService to pull back the necessary contact information related to the booking just created.
+    // Multiple contacts will result in multiple emails i.e. one email per contact.
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt:38]: Getting the owner email will be a later addition at present, so I'm getting the court, prison, probation and 3rd party contact details initially. I'll add it the owner email afterwards. `BookingContactsService.getBookingContacts(videoBookingId)` - just on the integration test, so coming very shortly.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/helper/EntityFactory.kt [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
 fun court(courtId: Long = 0, enabled: Boolean = true) = Court(
   courtId = courtId,
   code = "code",
-  description = "description",
+  description = "court description",
+  enabled = enabled,
+  notes = null,
+  createdBy = "Test",
+)
+
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/helper/EntityFactory.kt:16]: I must remember to add some of my test entity creation functions here.

**Review by timharrison-moj — COMMENTED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt:23]: Yes to comment. As for naming not sure, perhaps keep as is until you or I have an epiphany ...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt (hunk 1 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt:33]: No not at the moment, this is just the model from the search API.  Can add one though.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt:24]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt (hunk 1 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt:33]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt (hunk 2 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/Prison.kt:24]: Enabled == false, means courts/probation cannot self-serve but will accept prison bookings from DPS/AA
(for clarity)

**Review by timharrison-moj — APPROVED** [review]
LGTM
One minor clarification on the prison.enabled comment.

**Pull request merged**